modules/openxr/scene/openxr_composition_layer.cpp

```diff
@@ -328,9 +328,14 @@ void OpenXRCompositionLayer::set_use_android_surface(bool p_use_android_surface)
 
 	use_android_surface = p_use_android_surface;
 	if (use_android_surface) {
+        // It's possible that the layer provider is unregistered here (if previously invisible)
 		set_layer_viewport(nullptr);
 		openxr_layer_provider->set_use_android_surface(true, android_surface_size);
-	} else {
+        // ...and it may not be set up above because of viewport = null, android surface is false, so set it up again:
+        if (!registered && is_natively_supported() && openxr_session_running && is_inside_tree() && is_visible_in_tree()) {
```

Comment:
I'm starting to wonder if we should have a private `_should_register()` helper method for this? We've got this same condition in one other place, and another that could use it too (in `_on_openxr_session_begun()`) even if the current condition isn't exactly the same

Not necessary for this bug fix, though, we can clean it up later